boa/src/builtins/iterable/mod.rs

```diff
@@ -0,0 +1,154 @@
+use crate::builtins::string::string_iterator::StringIterator;
+use crate::object::GcObject;
+use crate::{
+    builtins::{
+        function::{BuiltInFunction, Function, FunctionFlags},
+        ArrayIterator,
+    },
+    object::{Object, PROTOTYPE},
+    property::Property,
+    BoaProfiler, Context, Value,
+};
+
+#[derive(Debug, Default)]
+pub struct IteratorPrototypes {
+    iterator_prototype: GcObject,
+    array_iterator: GcObject,
+    string_iterator: GcObject,
+}
+
+impl IteratorPrototypes {
+    pub fn init(ctx: &mut Context) -> Self {
+        let iterator_prototype = create_iterator_prototype(ctx);
+        Self {
+            iterator_prototype: iterator_prototype
+                .as_gc_object()
+                .expect("Iterator prototype is not an object"),
+            array_iterator: ArrayIterator::create_prototype(ctx, iterator_prototype.clone())
+                .as_gc_object()
+                .expect("Array Iterator Prototype is not an object"),
+            string_iterator: StringIterator::create_prototype(ctx, iterator_prototype)
+                .as_gc_object()
+                .expect("String Iterator Prototype is not an object"),
+        }
+    }
+
+    pub fn array_iterator(&self) -> GcObject {
+        self.array_iterator.clone()
+    }
+
+    pub fn iterator_prototype(&self) -> GcObject {
+        self.iterator_prototype.clone()
+    }
+
+    pub fn string_iterator(&self) -> GcObject {
+        self.string_iterator.clone()
+    }
+}
+
+/// CreateIterResultObject( value, done )
+///
+/// Generates an object supporting the IteratorResult interface.
+pub fn create_iter_result_object(ctx: &mut Context, value: Value, done: bool) -> Value {
+    let object = Value::new_object(Some(ctx.global_object()));
+    let value_property = Property::default().value(value);
+    let done_property = Property::default().value(Value::boolean(done));
+    object.set_property("value", value_property);
+    object.set_property("done", done_property);
+    object
+}
+
+/// Get an iterator record
+pub fn get_iterator(ctx: &mut Context, iterable: Value) -> Result<IteratorRecord, Value> {
```

Comment:
Any reason to not use `boa::Result`? Should be imported at the start, no?

Reply:
That's a good catch, thank you